### PR TITLE
Fix Keras imports.

### PR DESCRIPTION
**Pull request by reedwm**
# Description

Brief Description of the PR:

Recently, Keras changed how you must import internal symbols. Now, you must import private symbols from `keras.src` instead of `keras`, e.g. you must use `from keras.src.utils import tf_utils` instead of `from keras.utils import tf_utils`. See slide 12 of the [April Keras community meeting slides](https://drive.google.com/file/d/1uY4aJtYYnHOgmvBQO5yz_o36dsah4_8u/view) for details Because old versions of Keras, including the current stable version, require the old import form, this PR changes every file importing internal Keras symbols to try both import forms: first it tries importing the new version (with `keras.src`), and if that fails, it tries importing the old version (with `keras`)

Without this change, TF Addons cannot be imported with the latest `keras-nightly` pip package.

Fixes # (issue)

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [x] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*  Ran the official TensorFlow models with the updated TensorFlow Addons package, confirmed it worked correctly.

### CODE REVIEW

**Comment by reedwm** [comment]
CC @bhack

File: tensorflow_addons/optimizers/discriminative_layer_training.py [hunk 1]
```diff
@@ -21,8 +21,14 @@
 from tensorflow_addons.optimizers import KerasLegacyOptimizer
 from typeguard import typechecked
 
-from keras import backend
-from keras.utils import tf_utils
+try:
+    # New versions of Keras require importing from `keras.src` when
+    # importing internal symbols.
+    from keras.src import backend
```
Comment (sampathweb) [inline tensorflow_addons/optimizers/discriminative_layer_training.py:27]: Do we need this line?  Shouldn't `from keras import backend` work since its public API?

Comment (reedwm) [reply tensorflow_addons/optimizers/discriminative_layer_training.py:27]: Below, on line 144, `backend._current_graph()` is called. Since this is an internal symbol, it requires the `keras.src` import. `from keras import backend` will only allow public symbols to be accessed on `backend`, while `from keras.src import backend` allows all symbols from `backend.py` to be accessed.

Comment (sampathweb) [reply tensorflow_addons/optimizers/discriminative_layer_training.py:27]: Thanks for the details.  Sounds good.

**Review by sampathweb — DISMISSED** [review]

**Comment by bhack** [comment]
Are you using try/except as it is faster instead of checking the version?

**Comment by reedwm** [comment]
> Are you using try/except as it is faster instead of checking the version?

No, I use try-except because the issue only affects recent `keras-nightly` versions, and users with older `keras-nightly` versions need the old import. I don't think there's an easy way to check if the `keras-nightly` commit is newer than a certain commit.

**Comment by reedwm** [comment]
I think, but that version only refers to the stable version. The try-except will work on any given commit of Keras, which allows any version of `keras-nightly` to work. This is a fairly minor use case, so I can switch to a version check if you prefer.

**Comment by bhack** [comment]
Right, It is that keras-nightly is a quite ambiguous concept.

See my old point (2021) at:

https://github.com/keras-team/keras/issues/14867#issuecomment-876002266

I am ok with both. I am only worried that in the future users will be surprised by the wrong fallback import if the env is broken.

/cc @seanpmorgan what do you think?

**Comment by reedwm** [comment]
@bhack I switched to checking `tf.__version__` instead of using a try-except statement. I'm happy to switch it back if you prefer the try-except statement. The `tf.__version__` check means older version of `keras-nightly` will not be supported, but this is fine as users should expect to have to update the nightly TF and Keras packages if they also update TF-Addons.

The issue is causing various things to break, such as the official models, so it would be great if this could get merged quickly. Since @seanpmorgan hasn't responded yet, can you review this without @seanpmorgan's approval?

**Review by sampathweb — APPROVED** [review]

**Comment by bhack** [comment]
No problem, I am re-running two failed job. The I'll merge it.

**Review by bhack — APPROVED** [review]

**Comment by reedwm** [comment]
Thanks @bhack! Agreed, without CI coverage we cannot make any compatibility guarantees.
